src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -394,3 +412,1107 @@ def normalize(self, archive, logger):
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity', defaultDisplayUnit='mL/minute'
+        ),
+    )
+
+    pure_component = SubSection(section_def=PubChemPureSubstanceSection)
+
+    def normalize(self, archive, logger):
+        """
+        The normalizer will run for the subsection `PureSubstanceComponent` class.
+        A few exceptions are set here for reagents with ambiguous names or missing
+        entries in the PubChem database. A time.sleep(1) is set to prevent blocked IP
+        due to too many requests to the PubChem database.
+        If none is set, the normalizer will set the name of the component to be the
+        molecular formula of the substance.
+
+        Args:
+            archive (EntryArchive): The archive containing the section that is being
+            normalized.
+            logger ('BoundLogger'): A structlog logger.
+        """
+        super().normalize(archive, logger)
+
+        if self.name is None:
+            return
+        if self.name in ['C5-1', 'C6-1', 'nC5', 'nC6', 'Unknown', 'inert', 'P>=5C']:
+            return
+        elif self.name == 'n-Butene':
+            self.name = '1-butene'
+        elif self.name == 'MAN':
+            self.name = 'maleic anhydride'
+        elif '_' in self.name:
+            self.name = self.name.replace('_', ' ')
+
+        if self.name and self.pure_component is None:
+            import time
+
+            self.pure_component = PubChemPureSubstanceSection(name=self.name)
+            if self.name == 'propionic acid':
+                self.pub_chem_id = 1032
+                self.pure_component.iupac_name = 'propanoic acid'
+                self.pure_component.molecular_formula = 'C3H6O2'
+                self.pure_component.molecular_mass = 74.08
+                return
+            elif self.name in ['CO', 'carbon monoxide']:
+                self.pub_chem_id = 281
+                self.pure_component.iupac_name = 'carbon monoxide'
+                self.pure_component.molecular_formula = 'CO'
+                self.pure_component.molecular_mass = 28.01
+                self.pure_component.inchi = 'InChI=1S/CO/c1-2'
+                self.pure_component.inchi_key = 'UGFAIRIUMAVXCW-UHFFFAOYSA-N'
+                self.pure_component.cas_number = '630-08-0'
+                return
+            elif self.name in ['CO2', 'carbon dioxide']:
+                self.pub_chem_id = 280
+                self.pure_component.iupac_name = 'carbon dioxide'
+                self.pure_component.molecular_formula = 'CO2'
+                self.pure_component.molecular_mass = 44.01
+                self.pure_component.inchi = 'InChI=1S/CO2/c2-1-3'
+                self.pure_component.inchi_key = 'CURLTUGMZLYLDI-UHFFFAOYSA-N'
+                self.pure_component.cas_number = '124-38-9'
+                return
+            else:
+                time.sleep(1)
+                self.pure_component.normalize(archive, logger)
```

Comment:
Ah, reading this again I see that the super normalization is run after. Not sure what the `pure_component.normaliz()` does. You'll have to check if you need an extra super, can move the super before or leave it as is.

Reply:
One reason for also hardcoding the information here, was to save some extra requests from Pubchem since I had problems with hitting to many requests and beeing blocked. I should follow up on the open issue that we had which was supposed to fix the problem.

Reply:
also CO2 does not have a IUPAC name in Pubchem...

Reply:
I will for now. Hopefully I will come back to this, once the other issue is fixed.